### PR TITLE
feat(planning_validator): add invalid size error

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Add a trajectory size check in the planning validator. Before, the `planning_validator` does not perform any validation for trajectories with less than 3 trajectory points. After, the `planning_validator` raises an error for the no-enough-points trajectory.

## Related links

[TIERIV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-4973)

## Tests performed

1. change planning_validator publish policy to 1 or 2: https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/planning/scenario_planning/common/planning_validator/planning_validator.param.yaml/#L7
2. run psim
3. publish an empty trajectory and check if the control module won't show any error message on the terminal.

```sh
$ ros2 topic pub /planning/scenario_planning/motion_velocity_smoother/trajectory autoware_auto_planning_msgs/msg/Trajectory "{header: {frame_id: "base_link"}}"
```

**Before**

The trajectory_follower shows an error message, and the planning_evaluator process died.

image

**After**

Error raised only from the planing_validator.

image

## Notes for reviewers

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/planning_validator/src/planning_validator.cpp [hunk 1]
```diff
@@ -279,6 +294,11 @@ void PlanningValidator::validate(const Trajectory & trajectory)
   s.invalid_count = isAllValid(s) ? 0 : s.invalid_count + 1;
 }
 
+bool PlanningValidator::checkValidSize(const Trajectory & trajectory)
+{
+  return trajectory.points.size() < 2 ? false : true;
```
Comment (kosuke55) [inline planning/planning_validator/src/planning_validator.cpp:299]:
```suggestion
  return trajectory.points.size() >=2;
```

Comment (TakaHoribe) [reply planning/planning_validator/src/planning_validator.cpp:299]: cool. Thanks!

File: planning/planning_validator/src/planning_validator.cpp [hunk 2]
```diff
@@ -279,6 +294,11 @@ void PlanningValidator::validate(const Trajectory & trajectory)
   s.invalid_count = isAllValid(s) ? 0 : s.invalid_count + 1;
 }
 
+bool PlanningValidator::checkValidSize(const Trajectory & trajectory)
```
Comment (kosuke55) [inline planning/planning_validator/src/planning_validator.cpp:297]:
```suggestion
bool PlanningValidator::checkValidSize(const Trajectory & trajectory) const
```

File: planning/planning_validator/src/planning_validator.cpp [hunk 3]
```diff
@@ -429,10 +449,11 @@ bool PlanningValidator::checkValidDistanceDeviation(const Trajectory & trajector
 
 bool PlanningValidator::isAllValid(const PlanningValidatorStatus & s)
```
Comment (kosuke55) [inline planning/planning_validator/src/planning_validator.cpp:450]:
```suggestion
bool PlanningValidator::isAllValid(const PlanningValidatorStatus & s) const
```

File: planning/planning_validator/src/planning_validator.cpp (hunk 2 above)
Comment (kosuke55) [reply planning/planning_validator/src/planning_validator.cpp:297]: and we can add `const` to some functions in this class

File: planning/planning_validator/src/planning_validator.cpp (hunk 3 above)
Comment (kosuke55) [reply planning/planning_validator/src/planning_validator.cpp:450]: this is not modified part so no need to modify, but we can add `const` some functions in this class

File: planning/planning_validator/include/planning_validator/planning_validator.hpp [hunk 4]
```diff
@@ -64,6 +64,7 @@ class PlanningValidator : public rclcpp::Node
 
   void onTrajectory(const Trajectory::ConstSharedPtr msg);
 
+  bool checkValidSize(const Trajectory & trajectory);
```
Comment (kosuke55) [inline planning/planning_validator/include/planning_validator/planning_validator.hpp:67]:
```suggestion
  bool checkValidSize(const Trajectory & trajectory) const;
```

**Review by kosuke55 — APPROVED** [review]

File: planning/planning_validator/src/planning_validator.cpp (hunk 3 above)
Comment (TakaHoribe) [reply planning/planning_validator/src/planning_validator.cpp]: Done in https://github.com/autowarefoundation/autoware.universe/pull/6126/commits/992f7c123df3d8d5a019fab11c5d5e32f9f4237a

**Review by kosuke55 — APPROVED** [review]
LGTM again

**Pull request merged**